src/components/core/geoChart/geoChartDataSource.ts

```diff
@@ -93,49 +102,30 @@ function transformPushpinDataSource(
     return features;
 }
 
-export const pushpinSizeOptionsMap = {
-    min: {
-        "0.5x": 4,
-        "0.75x": 6,
-        default: 8,
-        "1.25x": 10,
-        "1.5x": 12,
-    },
-    max: {
-        "0.5x": 30,
-        "0.75x": 45,
-        default: 60,
-        "1.25x": 75,
-        "1.5x": 90,
-    },
-};
-
 // transform data value to pushpin size in pixel
-const transformPushpinSizeToPixel = (
+const calculatedSizeInPixel = (
     dataValue: number,
     minSize: number,
     maxSize: number,
-    pushpinSizeOptions: IGeoPushPinConfig,
+    pushpinSizeOptions: IGeoPointsConfig,
 ): number => {
     if (minSize === maxSize || dataValue === null) {
-        return pushpinSizeOptionsMap.min.default;
+        return PUSHPIN_SIZE_OPTIONS_MAP.min.default;
     }
-    const { minSize: minSizeOption = "default", maxSize: maxSizeOption = "default" } =
+    const { minSize: minSizeFromConfig = "default", maxSize: maxSizeFromConfig = "default" } =
         pushpinSizeOptions || {};
-    const minSizeOptionInPixel = pushpinSizeOptionsMap.min[minSizeOption];
-    const maxSizeOptionInPixel = pushpinSizeOptionsMap.max[maxSizeOption];
+    const minSizeInPixel = PUSHPIN_SIZE_OPTIONS_MAP.min[minSizeFromConfig];
+    const maxSizeInPixel = PUSHPIN_SIZE_OPTIONS_MAP.max[maxSizeFromConfig];
 
-    return (
-        ((dataValue - minSize) * (maxSizeOptionInPixel - minSizeOptionInPixel)) / (maxSize - minSize) +
-        minSizeOptionInPixel
-    );
+    return ((dataValue - minSize) * (maxSizeInPixel - minSizeInPixel)) / (maxSize - minSize) + minSizeInPixel;
 };
 
 export const createPushpinDataSource = (
     geoData: IGeoData,
-    groupNearbyPoints: boolean = true,
-    pushpinSizeOptions: IGeoPushPinConfig = {},
+    config?: IGeoConfig,
 ): mapboxgl.GeoJSONSourceRaw => {
+    const { points: pushpinSizeOptions = {} } = config || {};
+    const { groupNearbyPoints = true } = pushpinSizeOptions;
```

Comment:
Is groupNearbyPoints checked by default in configuration panel? If so, there is no issue here. Otherwise, it could cause inconsistency.

Reply:
it not cause of inconsistency,  it just the default value, if have config value, it will modify this default value.
On another hand, it useful for SDK user with default is true.